src/main/java/com/sparrowwallet/sparrow/control/DevicePane.java

```diff
@@ -794,8 +795,8 @@ private void sign() {
         }
     }
 
-    private void handleCardOperation(Service<?> service, ButtonBase operationButton, String operationDescription, boolean pinRequired, EventHandler<WorkerStateEvent> successHandler) {
-        if(pinRequired && pin.get().length() < 6) {
+    private void handleCardOperation(Service<?> service, ButtonBase operationButton, String operationDescription, boolean pinRequired, int pinMinLength, EventHandler<WorkerStateEvent> successHandler) {
```

Comment:
no need to add a parameter, you can access the `device` in this method to get the pinMinLength